### PR TITLE
[Bugfix] Support "starrocks.filter.query" for Spark SQL read

### LINKED ISSUE #91: 【Bug】error result for  Spark multi TMP view join
CREATE TEMPORARY VIEW spark_starrocks_1
           USING starrocks
           OPTIONS
           (
               "starrocks.table.identifier" = "test.score_board",
               "starrocks.filter.query" = "`name` = 'jhon' OR `name` = 'Lucy'",
               "starrocks.columns" = "id,name",
               "starrocks.fenodes" = "*******:8030",
               "starrocks.user" = "*****",
               "starrocks.password" = "******"
           );

企业微信截图_2da6cc4e-543f-4525-83a4-f90e8ba78d47

Assign to me, already solve this issue.

### PR DESCRIPTION
**Pull request by feihengye**
## What type of PR is this：
BugFix

## Which issues of this PR fixes ：
Fixes #91

### CODE REVIEW

File: src/main/scala/com/starrocks/connector/spark/sql/StarrocksRelation.scala [hunk 1]
```diff
@@ -83,7 +83,11 @@ private[sql] class StarrocksRelation(
     }
 
     if (filters != null && filters.length > 0) {
-      paramWithScan += (ConfigurationOptions.STARROCKS_FILTER_QUERY -> filterWhereClause)
+      val userFilters = paramWithScan.get(ConfigurationOptions.STARROCKS_FILTER_QUERY)
```
Comment (banmoy) [inline src/main/scala/com/starrocks/connector/spark/sql/StarrocksRelation.scala:86]: 1. what if the user configures the filter, but the framework's `filters` is null?
2. add a test to verify it. please refer to `ReadWriteITTest#testSql` for an example.

**Review by banmoy — COMMENTED** [review]
Thanks for the PR. I think it's ok to support user-defined filters in the configuration. Just curious about why not add filter in the join clause like
 `select * from spark_starrocks_4 a join spark_starrocks_4 b on a.name = b.name where a.name = 'jhon' or a.name = 'Lucy'`
The filter will be pushed down to the source automatically, and it seems more flexible. Defining the filter in the configuration is generally used for DataFrame program.

**Comment by banmoy** [comment]
Please deal with the DCO and CLA check failure

**Comment by feihengye** [comment]
> Thanks for the PR. I think it's ok to support user-defined filters in the configuration. Just curious about why not add filter in the join clause like `select * from spark_starrocks_4 a join spark_starrocks_4 b on a.name = b.name where a.name = 'jhon' or a.name = 'Lucy'` The filter will be pushed down to the source automatically, and it seems more flexible. Defining the filter in the configuration is generally used for DataFrame program.

User create tmp view and config the push down filters already， it is a table now.

File: src/main/scala/com/starrocks/connector/spark/sql/StarrocksRelation.scala (hunk 1 above)
Comment (feihengye) [reply src/main/scala/com/starrocks/connector/spark/sql/StarrocksRelation.scala:86]: 1. use old config, ConfigurationOptions.STARROCKS_FILTER_QUERY
2. ok

**Comment by banmoy** [comment]
LGTM. Thanks.

**Pull request merged**